bionic/protocols.py

```diff
@@ -599,6 +599,69 @@ def read(self, path):
         return single_element(path.iterdir())
 
 
+# This protocol allows us to hash sets deterministically.
+# It would be preferable if PicklableProtocol could install a special handler for sets,
```

Comment:
I'm not 100% sure if this will work, but is it possible to use [`dispatch_table`](https://python.readthedocs.io/en/stable/library/pickle.html#pickle.Pickler.dispatch_table) to install a handler for sets?

Reply:
Unfortunately not. I don't think either of the override mechanisms works for built-in types. (The pure Python implementation of `pickle` has a separate undocumented dispatch table for built-in types that you can modify, but in the faster C implementation (which we use) the behavior for sets is hardcoded.)

Reply:
I saw the implementation too and now see what you mean by it. Technically, we can subclass the pickler in Bionic and modify `self.dispatch` variable to override set behavior, but that seems risky given that variable isn't documented. I guess this is fine then.

Reply:
Yeah, and the subclassing approach only works for the pure Python implementation, so all of our pickling would get slower.